config/routes.rb

```diff
@@ -3,6 +3,12 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root 'users#index'
+
+  resources :users do 
+     resources :posts, only: [:index, :show]
+      post '/posts', to: 'posts#index'
+    # get '/posts/:id', to: 'posts#show'
+  end
+  resources :posts, only: [:show]
```

Comment:
You don't need to specify the show route for posts outside the users, since it is already nested under user. Defining a  post route outside the `resources :users do` block, will allow accessing individual posts independently of the associated user which should not be the case. Kindly remove it.

Reply:
Done, as advised